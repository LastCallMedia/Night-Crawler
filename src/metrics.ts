// @flow

export interface Metric {
  displayName: string;
  level: number;
  value: number;
  toString(): string;
}

export class Number implements Metric {
  displayName: string;
  level: number;
  value: number;
  constructor(label: string, level: number, value: number) {
    this.displayName = label;
    this.level = level;
    this.value = value;
  }
  toString() {
    return `${this.value}`;
  }
}

export class Percent extends Number {
  toString() {
    return `${Math.round(this.value * 100)}%`;
  }
}

export class Milliseconds extends Number {
  toString() {
    return `${Math.round(this.value)}ms`;
  }
}
