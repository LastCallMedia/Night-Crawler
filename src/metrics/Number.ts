import { Metric } from '../analysis';

export default class Number implements Metric {
  displayName: string;
  level: number;
  value: number;
  constructor(label: string, level: number, value: number) {
    this.displayName = label;
    this.level = level;
    this.value = value;
  }
  toString() {
    return `${this.value}`;
  }
}
