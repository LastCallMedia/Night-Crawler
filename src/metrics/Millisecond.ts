import Number from './Number';

export default class Milliseconds extends Number {
  toString() {
    return `${Math.round(this.value)}ms`;
  }
}
