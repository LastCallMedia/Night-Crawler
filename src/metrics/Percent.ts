import Number from './Number';

export default class Percent extends Number {
  toString() {
    return `${Math.round(this.value * 100)}%`;
  }
}
