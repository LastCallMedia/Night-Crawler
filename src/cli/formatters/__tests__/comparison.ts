import ConsoleComparisonFormatter from '../comparison';
import Analysis from '../../../analysis';
import { Number } from '../../../metrics';

describe('Console Comparison Formatter', function() {
  it('Should compare metrics from two analyses', function() {
    var a = new Analysis('test', new Date());
    a.addMetric('ok', new Number('OK metric', 0, 0));
    a.addMetric('warn', new Number('WARN metric', 1, 2));
    a.addMetric('err', new Number('ERR metric', 2, 5));
    var b = new Analysis('test', new Date());
    b.addMetric('ok', new Number('OK metric', 0, 0));
    b.addMetric('warn', new Number('WARN metric', 1, 2));
    b.addMetric('err', new Number('ERR metric', 2, 5));
    expect(new ConsoleComparisonFormatter().format([a, b])).toMatchSnapshot();
  });
  it('Should output "no results" if a report has no metrics', function() {
    var a = new Analysis('test', new Date());
    var b = new Analysis('test2', new Date());
    expect(new ConsoleComparisonFormatter().format([a, b])).toMatchSnapshot();
  });
});
