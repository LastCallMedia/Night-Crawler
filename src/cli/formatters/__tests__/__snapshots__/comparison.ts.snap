// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Console Comparison Formatter Should compare metrics from two analyses 1`] = `
"╔═════════════╤═══════════╤═══════════╗
║ Name        │ Report #1 │ Report #2 ║
╟─────────────┼───────────┼───────────╢
║ OK metric   │ 0         │ 0 (0%)    ║
╟─────────────┼───────────┼───────────╢
║ WARN metric │ [33m2[39m         │ [33m2[39m (100%)  ║
╟─────────────┼───────────┼───────────╢
║ ERR metric  │ [31m5[39m         │ [31m5[39m (100%)  ║
╚═════════════╧═══════════╧═══════════╝
"
`;

exports[`Console Comparison Formatter Should output "no results" if a report has no metrics 1`] = `"[33mNo Results[39m"`;
